### PR TITLE
feat(annoucement-of-death): Firearms 

### PR DESCRIPTION
**Pull request by joningi98**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced firearm applicant information handling in the Announcement of Death application, including fields for national ID, name, phone, and email.
  - Added email and SMS notifications related to firearm applicants.
  - Integrated new messages and form sections for firearm-related details.

- **Enhancements**
  - Improved form structure with new sub-sections and fields for firearm details.
  - Enhanced data schema to support firearm applicant information.
  
- **Notifications**
  - Added functionality to notify firearm applicants via email and SMS.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206830831929263

### CODE REVIEW

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

File: libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/announcement-of-death.service.ts [hunk 1]
```diff
@@ -23,19 +23,27 @@ import {
...
 export class AnnouncementOfDeathService extends BaseTemplateApiService {
   constructor(
     @Inject(LOGGER_PROVIDER) private logger: Logger,
     private readonly syslumennService: SyslumennService,
+    private readonly sharedTemplateAPIService: SharedTemplateApiService,
   ) {
     super(ApplicationTypes.ANNOUNCEMENT_OF_DEATH)
   }
 
+  private log(lvl: 'error' | 'info', message: string, meta: unknown) {
```
Comment (juni-haukur) [inline libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/announcement-of-death.service.ts:43]: is this being used?

File: libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/emailGenerators/firearmApplicantNotification.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    application,
+    options: { email = { sender: 'Ísland.is', address: 'no-reply@island.is' } },
+  } = props
+
+  const firearmApplicant = application.answers
+    .firearmApplicant as AnnouncementOfDeathAnswers['firearmApplicant']
+
+  if (!firearmApplicant) throw new Error('Firearm applicant was undefined')
+
+  const subject = 'Tilkynning um vörslu skotvopna'
```
Comment (juni-haukur) [inline libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/emailGenerators/firearmApplicantNotification.ts:25]: translatable?

File: libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/emailGenerators/firearmApplicantNotification.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        },
+        {
+          component: 'Heading',
+          context: { copy: subject },
+        },
+        {
+          component: 'Copy',
+          context: {
+            copy:
+              `<span>Góðan dag,</span><br/><br/>` +
```
Comment (juni-haukur) [inline libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/emailGenerators/firearmApplicantNotification.ts:57]: translatable?

File: libs/application/templates/announcement-of-death/src/forms/draft/subSectionFirearms.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  buildCustomField,
+  buildMultiField,
+  buildRadioField,
+  buildSubSection,
+} from '@island.is/application/core'
+import { m } from '../../lib/messages'
+
+export const subSectionFirearms = buildSubSection({
+  id: 'firearmsStep',
+  title: 'Skotvopn',
```
Comment (juni-haukur) [inline libs/application/templates/announcement-of-death/src/forms/draft/subSectionFirearms.ts:11]: Translatable?

**Review by juni-haukur — APPROVED** [review]

**Pull request merged**